internal/workload/service/systemd.go

```diff
@@ -0,0 +1,168 @@
+package service
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+
+	"git.sr.ht/~spc/go-log"
+)
+
+const (
+	DefaultUnitsPath      = "/etc/systemd/system/"
+	DefaultRestartTimeout = 15
+	DefaultServiceSuffix  = ".service"
+	DefaultServicePrefix  = "pod-"
+)
+
+type Systemd struct {
+	Name         string            `json:"name"`
+	RestartSec   int               `json:"restartSec"`
+	Path         string            `json:"path"`
+	Units        []string          `json:"units"`
+	UnitsContent map[string]string `json:"-"`
+}
+
+type SystemdManager struct {
+	svcFilePath string
+	lock        sync.RWMutex
+	services    map[string]*Systemd
+}
+
+func NewSystemdManager(configDir string) (*SystemdManager, error) {
+	services := make(map[string]*Systemd)
+
+	servicePath := path.Join(configDir, "services.json")
+	servicesJson, err := ioutil.ReadFile(servicePath)
+	if err == nil {
+		err := json.Unmarshal(servicesJson, &services)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &SystemdManager{svcFilePath: servicePath, services: services, lock: sync.RWMutex{}}, nil
+}
+
+func (mgr *SystemdManager) Add(svc *Systemd) error {
+	mgr.lock.Lock()
+	defer mgr.lock.Unlock()
+
+	mgr.services[svc.Name] = svc
+
+	return mgr.write()
+}
+
+func (mgr *SystemdManager) Get(name string) *Systemd {
+	mgr.lock.Lock()
+	defer mgr.lock.Unlock()
+
+	return mgr.services[name]
+}
+
+func (mgr *SystemdManager) Remove(svc *Systemd) error {
+	mgr.lock.Lock()
+	defer mgr.lock.Unlock()
+
+	delete(mgr.services, svc.Name)
+
+	return mgr.write()
+}
+
+func (mgr *SystemdManager) write() error {
+	svcJson, err := json.Marshal(mgr.services)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(mgr.svcFilePath, svcJson, 0640)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func NewSystemd(name string, units map[string]string) (*Systemd, error) {
+	path, err := exec.LookPath("systemctl")
+	if err != nil {
+		return nil, err
+	}
+
+	var unitNames []string
+	for unit := range units {
+		unitNames = append(unitNames, unit)
+	}
+
+	return &Systemd{
+		Name:         name,
+		RestartSec:   DefaultRestartTimeout,
+		Path:         path,
+		Units:        unitNames,
+		UnitsContent: units,
```

Comment:
Is there a need to define `Requires` section to list the dependencies? (is `podman.socket` a must or `nftables.service`) ?

Reply:
Another question about dependencies, this is out of the scope for this PR since we don't yet support it, but for deployments-spec instead of pod-spec for the podman-kube-play, is there any relation between the generated service?

Reply:
> Is there a need to define Requires section to list the dependencies? (is podman.socket a must or nftables.service) ?

`podman.socket` is a must for the `device-worker` itself, it won't start without it. Same for `nft`.
Podman pods, have it's own dependecies, which are generated from the podman code itself, like the containers, which pod consist of. But those are generated by the podman, no need to store it here. We just store all the units, which must be deleted, after the pod is removed.

> Another question about dependencies, this is out of the scope for this PR since we don't yet support it, but for deployments-spec instead of pod-spec for the podman-kube-play, is there any relation between the generated service?

We would just need to store, which services(units) must be removed in case deployment is removed. We have relation pod<->containers, so we would have relation deployment<->pods<->containers.